packages/sync/src/PersistentStore.ts

```diff
@@ -9,3 +8,199 @@ export interface PersistentStore<T> {
 }
 
 export type PersistedData = string | null | object;
+
+export class StorageWrapper implements PersistentStore<PersistedData> {
+  public storage: PersistentStore<PersistedData>;
+  private crypto = window.crypto;
+  private masterKey: CryptoKey | undefined;
+  private password: string;
+
+  constructor(storage: PersistentStore<PersistedData>, encryptionPassword: string) {
+    this.password = encryptionPassword;
+    this.storage = storage;
+    this.init();
+  }
+
+  /**
+   * Retrieves an encrypted item from storage.
+   * First extracts the initialisation vector that was previously prepended to the stored encrypted data
+   * @param key the key to use to retreive your value
+   */
+  public getItem(key: string) {
+    const retrieved = this.storage.getItem(key);
+    if (retrieved) {
+      if (this.masterKey) {
+        const cipherTextWithIv = retrieved.toString();
+        const cipherText = cipherTextWithIv.slice(cipherTextWithIv.indexOf(":") + 1);
+        const iv = this.strToArray(cipherTextWithIv.substr(0, cipherTextWithIv.indexOf(":")));
+        this.crypto.subtle.decrypt({ name: "AES-GCM", iv },
```

Comment:
it might be better to move L33-36 into its own function to make the `getItem` function easier to read.